Update docs to replace filter with relationships

svanharmelen opened the pull request:
This change is done inline with the changes in https://github.com/hashicorp/atlas/pull/6303

nfagerlund commented:
Oh man, I remember wondering whether that was a docs error or what. Hooray for making it consistent!

armchairlinguist commented on content/source/docs/enterprise/api/variables.html.md:
```diff
@@ -46,14 +48,21 @@ Key path                    | Type   | Default | Description
+      "workspace": {
+        "data": {
+          "type":"workspaces",
+          "id":"ws-123456789"
```
This id should look like a real workspace ID.

svanharmelen commented:
Thanks for the feedback @armchairlinguist! I updated the workspace ID and added a deprecation warning. The `JSONAPI` specs do not offer a way to deprecate API's or API fields, so indeed adding this to the docs seems like the right thing to do 👍

armchairlinguist commented on content/source/docs/enterprise/api/variables.html.md:
```diff
@@ -22,16 +22,25 @@ This POST endpoint requires a JSON object with the following properties as a req
+`data.relationships.organization.data.type` | string |         | Must be `"organizations"`.
+`data.relationships.organization.data.id`   | string |         | The ID (which is the name) of the organization that owns the workspace.
+
+**Deprecation warning**: The following properties are deprecated and will be removed from feature versions of the API!
```
This looks like a great add. I'd replace "feature" with "future", and note something like "`filter` properties have been replaced by `relationships`.

svanharmelen replied:
That's indeed a typo, will update that and add a word about the `filter` <-> `relationships` change as well. Thought that would be obvious, doesn't hurt to write it down 👍 Thx!

svanharmelen commented:
@armchairlinguist tweaked it a bit more... Could you double check before I merge the PR?
